File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        map<int,int> hash;
        int n = nums.size();
        
        for(int i=0;i<n;i++){
            hash[nums[i]] = i;
        }
        
        for(int i=0;i<n;i++){
            int rem = target-nums[i];
            
            if(hash.find(rem)!=hash.end() && i!=hash[rem]){
                return {i,hash[rem]};
            }
        }
        
        return {};
        
    }
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution
{
    public int[] TwoSum(int[] nums, int target)
    {
        var dct = new Dictionary<int, int>();

        for(int i = 0; i < nums.Length; i++)
        {
            int rem = target-nums[i];

            if(!dct.ContainsKey(rem))
            {
                dct[nums[i]] = i;
            }
            else
            {
                return new int[2]{i, dct[rem]};
            }
        }
        return new int[]{};
    }
}